docs/rules/max-len.md

```diff
@@ -0,0 +1,124 @@
+---
+pageClass: "rule-details"
+sidebarDepth: 0
+title: "markdown-preferences/max-len"
+description: "enforce maximum length for various Markdown entities"
+---
+
+# markdown-preferences/max-len
+
+> enforce maximum length for various Markdown entities
+
+- ❗ <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
+- ⚙️ This rule is included in `plugin.configs.standard`.
+
+## 📖 Rule Details
+
+This rule enforces a configurable maximum length for different Markdown entities such as headings, paragraphs, list items, blockquotes, table rows, and optionally code blocks.
+
+Overly long Markdown elements reduce readability, make diffs harder to review, and complicate maintenance. By enforcing reasonable maximum lengths for various entities, documentation remains cleaner, easier to navigate, and more consistent across the project.
+
+<!-- prettier-ignore-start -->
+
+<!-- eslint-skip -->
+
+```md
+<!-- eslint markdown-preferences/max-len: ['error', { heading: 80, paragraph: 120 }] -->
+
+<!-- ✓ GOOD -->
+
+## This heading is within the limit
+
+This is a paragraph that fits comfortably within the specified maximum length limit.
+
+- A list item that is short enough
+
+<!-- ✗ BAD -->
+
+## This is a very long heading that exceeds the default eighty character maximum length
+
+This is an extremely long paragraph that goes on and on and on well beyond the one hundred and twenty character maximum length limit and should be broken up.
+
+- This is a very long list item that definitely exceeds the default one hundred and twenty character maximum length limit and should be reported
+```
+
+<!-- prettier-ignore-end -->
+
+## 🔧 Options
+
+This rule accepts an object with the following properties:
+
+```json
+{
+  "markdown-preferences/max-len": [
+    "error",
+    {
+      "heading": 80,
+      "paragraph": 120,
+      "listItem": 120,
+      "blockquote": 120,
+      "tableRow": 120,
+      "codeBlock": null,
+      "ignoreUrls": true
+    }
+  ]
+}
+```
+
+### Entity-Specific Options
+
+- `heading` (default: `80`): Maximum line length for headings
+- `paragraph` (default: `120`): Maximum line length for paragraphs
+- `listItem` (default: `120`): Maximum line length for list items
+- `blockquote` (default: `120`): Maximum line length for blockquotes
```

Comment:
What about headers within blockquotes?
How about making it possible to set a header like `{header: 80}` in the `blockquote` optiopn?